### PR TITLE
[ML] Introduction disk usage parameter for data frames

### PR DESCRIPTION
**Pull request by valeriy42**
This PR introduces an optional json specification parameter `disk_usage_allowed`, which is `false` by default. This parameter controls whether or not the system can overflow data frame to disk. 

Relates elastic/ml-team#168

### CODE REVIEW

**Comment by valeriy42** [comment]
retest

**Comment by valeriy42** [comment]
recheck

File: lib/api/CDataFrameAnalysisRunner.cc [hunk 1]
```diff
@@ -63,6 +63,11 @@ void CDataFrameAnalysisRunner::computeAndSaveExecutionStrategy() {
         if (memoryUsage <= memoryLimit) {
             break;
         }
+        // if we are not allowed to spill over to disk than only one partition is possible
```
Comment (dimitris-athanasiou) [inline lib/api/CDataFrameAnalysisRunner.cc]: nit: typo `than` -> `then`?

**Comment by valeriy42** [comment]
retest

**Comment by valeriy42** [comment]
retest

**Comment by droberts195** [comment]
retest

**Comment by droberts195** [comment]
On this one the Windows failure is _not_ due to the CI infra:

```
CDataFrameAnalysisRunner.cc(67,13): error C2065: 'not': undeclared identifier
        if (not m_Spec.diskUsageAllowed()) {
            ^
```

We’ve used `== false` as an alternative to `!` elsewhere.

**Pull request merged**